src/xbot/common/controls/sensors/Gamepad.java

```diff
@@ -1,52 +1,114 @@
 package xbot.common.controls.sensors;
 
+import edu.wpi.first.wpilibj.GamepadBase;
 import edu.wpi.first.wpilibj.GenericHID;
+import edu.wpi.first.wpilibj.GenericHID.Hand;
 import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj.XboxController;
 import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter;
 import xbot.common.controls.sensors.wpi_adapters.GamepadJoystickWpiAdapter.GamepadComponent;
+import xbot.common.math.XYPair;
 
 public class Gamepad implements XGamepad {
     
-    final GenericHID internalHID;
-    
-    final XJoystick leftStick;
-    final XJoystick rightStick;
-    final XJoystick dpad;
-    final XJoystick leftTrigger;
-    final XJoystick rightTrigger;
+    final XboxController internalGamepad;
     
     public Gamepad(int port) {
-        internalHID = new Joystick(port);
         
-        leftStick = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.LeftJoystick);
-        rightStick = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.RightJoystick);
-        dpad = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.DPad);
-        leftTrigger = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.LeftTrigger);
-        rightTrigger = new GamepadJoystickWpiAdapter(internalHID, GamepadComponent.RightTrigger);
-    }
-
-    @Override
-    public XJoystick getLeftStick() {
-        return leftStick;
+        internalGamepad = new XboxController(port);
     }
 
     @Override
-    public XJoystick getRightStick() {
-        return rightStick;
-    }
+    //Joysticks---------------------------------------------------------------------------------------------
+   public double getLeftStickX(){
+       
+       return internalGamepad.getX(Hand.kLeft);
+   }
     
-    @Override
-    public XJoystick getDpad() {
-        return dpad;
+    public double getRightStickX(){
+        
+        return internalGamepad.getX(Hand.kRight);
     }
+    
+   public double getLeftStickY(){
+       
+       return internalGamepad.getY(Hand.kLeft);
+   }
+   
+   public double getRightStickY(){
```

Comment:
Add `@Override` to all these methods that implement the interface. That way, errors will occur if any of these methods are wrong. Same with the mock impl.